modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -141,10 +147,16 @@ def from_partitions(
     from modin.core.execution.dispatching.factories.dispatcher import FactoryDispatcher
 
     factory = FactoryDispatcher.get_factory()
-
-    partition_class = factory.io_cls.frame_cls._partition_mgr_cls._partition_class
+    assert factory is not None
+    assert factory.io_cls is not None
+    assert factory.io_cls.frame_cls is not None
+    assert factory.io_cls.frame_cls._partition_mgr_cls is not None
```

Comment:
These shouldn't be `None`. Does mypy complain that they could be `None`?

Reply:
Hi @mvashishtha this is the complaints from mypy
```
modin/distributed/dataframe/pandas/partitions.py:157:27: error: "None" has no attribute "_partition_mgr_cls"  [attr-defined]

modin/distributed/dataframe/pandas/partitions.py:158:23: error: "None" has no attribute "_partition_mgr_cls"  [attr-defined]
```
Do you think we can ignore it?

Reply:
It may be that we need to add mypy hints to the FactoryDispatcher.

Reply:
@rosdyana could you please add the assertions back so we stop getting these failures, and also add a comment like this:

# TODO(https://github.com/modin-project/modin/issues/5127): Remove these assertions once the dependencies of this function all have types.